packages/core/src/lib/lib/interactivity.ts

```diff
@@ -68,27 +75,23 @@ export const useEventRaycast = (
   onDestroy(unsubscribePointer)
 
   const onEvent = (e: MouseEvent | PointerEvent) => {
+    const eventType = e.type as keyof ThreltePointerEventMap
     e.preventDefault()
     ctx.pointerOverCanvas.set(true)
     renderCtx.pointerInvalidated = true
     setPointerFromEvent(ctx, e)
 
-    if (rootCtx.interactiveObjects.size === 0 || rootCtx.raycastableObjects.size === 0) return
-    const intersects = runRaycaster(
-      rootCtx,
-      pointer,
-      camera,
-      Array.from(rootCtx.raycastableObjects)
-    )
-    if (intersects.length > 0 && rootCtx.interactiveObjects.has(intersects[0].object)) {
-      getThrelteUserData(intersects[0].object).eventDispatcher?.(
-        e.type as keyof ThreltePointerEventMap,
-        {
-          ...intersects[0],
-          event: e
-        }
-      )
+    const closestIntersection = getClosestIntersection(rootCtx, pointer, camera)
+    if (eventType === 'pointerdown' && e.button === MOUSE.LEFT) {
```

Comment:
Did you check if that check also working on mobile? Do you do that check because `pointerdown` is also called for the right click?

Reply:
It works with mobile/tap emulation in Chrome, at least. Yeah, that was the thought behind it - although I suppose it might not be strictly necessary and would only affect if you're doing something weird like holding several mouse buttons simultaneously. I think it's theoretically more correct to have the check, but don't mind removing it if you think it's cleaner without.

Reply:
R3F handles this by checking the object on pointerdown only and I think this is the better because solution because it is more generic and has less unknown side effects: https://github.com/pmndrs/react-three-fiber/blob/master/packages/fiber/src/core/events.ts#L418-L422

Reply:
Fixed!